AddNewPod.md

```diff
@@ -0,0 +1,41 @@
+# Adding a New Firebase CocoaPod
+
+## Introduction
+
+The Firebase build is driven by the contents of a podspec. It is helpful to
+use an existing podspec as a template when starting a new pod.
+
+## Podspec attributes
+
+See the [Podspec Syntax Reference](https://guides.cocoapods.org/syntax/podspec.html) for
+detailed instructions. Some Firebase specific guidance below:
+
+* `s.deployment_target` - Ideally should include ios, osx, and tvos. See
+[FirebaseCore.podspec](FirebaseCore.podspec) for the current Firebase minimum version settings.
+
+* `s.static_framework` - By default, Firebase pods should be static frameworks.
+
+* `s.dependency` - Dependencies on other Firebase pods should allow minor version updates - 
+like `s.dependency 'FirebaseCore', '~> 6.0'`
+
+* `s.pod_target_xcconfig` - Add any specific build settings. For portability, any Firebase
+pod with other Firebase dependencies should build for c99 -
+`'GCC_C_LANGUAGE_STANDARD' => 'c99'`. The pod's version should be passed in as a #define
+for FIRComponent registration. See examples of setting `GCC_PREPROCESSOR_DEFINITIONS`.
+
+* `s.test_spec` should be used for defining all unit and integration test suites.
+
+
+## Directory Structure
+
+The Firebase library `Foo` should be defined in `FirebaseFoo.podspec`. All of its
+contents should be in the `FirebaseFoo` directory.
+
+* FirebaseFoo/Source - All source. Directory structure is up to the library owner.
+* FirebaseFoo/Public - Public Headers.
```

Comment:
For C++ targets this isn't a good layout (and is counter to Google C++ style, where all directory and filenames should be lowercase). SPM allows for specifying these folders and (I've read that) it implicitly picks up names like `src` and `include`.

I just want to confirm that even though we're not specifically encouraging this in other modules, sources in `src`, public headers in `include`, and similar will be OK there.

Reply:
I defer to you for the C++ adaptation.